src/EditorFeatures/VisualBasicTest/AddMissingImports/VisualBasicAddMissingImportsFeatureServiceTests.vb

```diff
@@ -0,0 +1,137 @@
+            Dim code = "
+Class C
+    Dim foo As [|D|]
+End Class
+
+Namespace A
+    Public Class D
+    End Class
+End Namespace
+"
+
+            Dim expected = "
+Imports A
+
+Class C
+    Dim foo As D
+End Class
+
+Namespace A
+    Public Class D
+    End Class
+End Namespace
+"
+
+            Await AssertDocumentChangedAsync(code, expected).ConfigureAwait(False)
+        End Function
+
+        <Fact>
+        Public Async Function AddMissingImports_DocumentUnchanged_SpanContainsAmbiguousImports() As Task
+            Dim code = "
+Class C
+    Dim foo As [|D|]
+End Class
+
+Namespace A
+    Public Class D
+    End Class
+End Namespace
+
+Namespace B
+    Public Class D
+    End Class
+End Namespace
+"
+
+            Await AssertDocumentUnchangedAsync(code).ConfigureAwait(False)
+        End Function
+
+        Private Async Function AssertDocumentUnchangedAsync(initialMarkup As String) As Task
+            Using workspace = TestWorkspace.CreateVisualBasic(initialMarkup)
+
+                Dim diagnosticAnalyzerService = InitializeDiagnosticAnalyzerService(workspace)
+
+                Dim addMissingImportsService = New VisualBasicAddMissingImportsFeatureService(diagnosticAnalyzerService)
+
+                Dim hostDocument = workspace.Documents.First()
+                Dim documentId = hostDocument.Id
+                Dim textSpan = hostDocument.SelectedSpans.First()
+
+                Dim document = workspace.CurrentSolution.GetDocument(documentId)
+
+                Dim newProject = Await addMissingImportsService.AddMissingImportsAsync(document, textSpan, CancellationToken.None).ConfigureAwait(False)
+                Dim newDocument = newProject.GetDocument(documentId)
+
+                Assert.Equal(document, newDocument)
+            End Using
+        End Function
+
+        Private Async Function AssertDocumentChangedAsync(initialMarkup As String, expectedMarkup As String) As Task
+            Using workspace = TestWorkspace.CreateVisualBasic(initialMarkup)
+
+                Dim diagnosticAnalyzerService = InitializeDiagnosticAnalyzerService(workspace)
+
+                Dim addMissingImportsService = New VisualBasicAddMissingImportsFeatureService(diagnosticAnalyzerService)
+
+                Dim hostDocument = workspace.Documents.First()
+                Dim documentId = hostDocument.Id
+                Dim textSpan = hostDocument.SelectedSpans.First()
+
+                Dim document = workspace.CurrentSolution.GetDocument(documentId)
+
+                Dim newProject = Await addMissingImportsService.AddMissingImportsAsync(document, textSpan, CancellationToken.None).ConfigureAwait(False)
+                Dim newDocument = newProject.GetDocument(documentId)
+
+                Assert.NotEqual(document, newDocument)
+
+                Dim Text = Await newDocument.GetTextAsync().ConfigureAwait(False)
+
+                Assert.Equal(expectedMarkup, Text.ToString())
```

Comment:
this code looks very similar to code in the C# side of things.  Is there a reason we need to duplicate this?